### ISSUE TITLE
4.3.12 timing is wrong on some schedule steps

### ISSUE THREAD

**Issue by CDoughty08** [Hexagon/croner#110]
OS: RHEL 8/7
Node: 16.15.0

Running the below on 4.3.11 gives expected results (1 run per 5 seconds)
```ts
const cron = new Cron('*/5 * * * * *');
cron.schedule(() => console.log(`running: ${new Date()}:${Date.now()}`));
```
Output:
```
running: Wed Jun 15 2022 15:00:25 GMT-0500 (Central Daylight Time):1655323225001
running: Wed Jun 15 2022 15:00:30 GMT-0500 (Central Daylight Time):1655323230001
running: Wed Jun 15 2022 15:00:35 GMT-0500 (Central Daylight Time):1655323235001
running: Wed Jun 15 2022 15:00:40 GMT-0500 (Central Daylight Time):1655323240000
running: Wed Jun 15 2022 15:00:45 GMT-0500 (Central Daylight Time):1655323245000
running: Wed Jun 15 2022 15:00:49 GMT-0500 (Central Daylight Time):1655323249999
running: Wed Jun 15 2022 15:00:50 GMT-0500 (Central Daylight Time):1655323250001
running: Wed Jun 15 2022 15:00:55 GMT-0500 (Central Daylight Time):1655323255001
running: Wed Jun 15 2022 15:00:59 GMT-0500 (Central Daylight Time):1655323259999
running: Wed Jun 15 2022 15:01:05 GMT-0500 (Central Daylight Time):1655323265000
```

However, 4.3.12
gives the below output, this seems to happen with all schedules at some random frequency.
```
running: Wed Jun 15 2022 14:56:40 GMT-0500 (Central Daylight Time):1655323000001

---- 14:56:45 is ran almost 5 seconds late immediately followed by the 14:56:50 run
**running: Wed Jun 15 2022 14:56:49 GMT-0500 (Central Daylight Time):1655323009997**
**running: Wed Jun 15 2022 14:56:50 GMT-0500 (Central Daylight Time):1655323010001**

running: Wed Jun 15 2022 14:56:55 GMT-0500 (Central Daylight Time):1655323015000
running: Wed Jun 15 2022 14:57:00 GMT-0500 (Central Daylight Time):1655323020000
running: Wed Jun 15 2022 14:57:05 GMT-0500 (Central Daylight Time):1655323025001
running: Wed Jun 15 2022 14:57:10 GMT-0500 (Central Daylight Time):1655323030001
running: Wed Jun 15 2022 14:57:15 GMT-0500 (Central Daylight Time):1655323035000

---- 14:57:20 is ran almost 5 seconds late immediately followed by the 14:57:25 run
**running: Wed Jun 15 2022 14:57:24 GMT-0500 (Central Daylight Time):1655323044999**
**running: Wed Jun 15 2022 14:57:25 GMT-0500 (Central Daylight Time):1655323045000**
```

**Comment by Hexagon** [comment]
Oh, thanks for reporting this. Fixed in `4.3.13` by reverting changes in `4.3.12`. Reopening #101.

**Comment by Hexagon** [comment]
@CDoughty08 tried to re-apply original fix from 4.3.12 in 4.3.14, if possible - verify that the problem isn't back in 4.3.14
